plugins/modules/consul_role.py

```diff
@@ -0,0 +1,590 @@
+          - Either this or O(policies[].id) must be specified.
+        type: str
+      id:
+        description:
+          - The id of the policy to attach to this role; see M(community.general.consul_policy) for more info.
+          - Either this or O(policies[].name) must be specified.
+        type: str
+  service_identities:
+    type: list
+    elements: dict
+    description:
+      - List of service identities to attach to the role.
+      - Each element must have a "name" and optionally a "datacenters" list of datacenters the effective policy is valid within.
+      - An empty datacenters list allows all datacenters
+      - If the parameter is an empty array ([]), any node identities assigned will be unassigned.
+    required: false
+  node_identities:
+    type: list
+    elements: dict
+    description:
+      - List of node identities to attach to the role.
+      - Each element must have a "name" and optionally a "datacenter" the effective policy is valid in.
+      - An empty datacenter allows all datacenters.
+      - If the parameter is an empty array ([]), any node identities assigned will be unassigned.
+    required: false
+  host:
+    description:
+      - Host of the consul agent, defaults to V(localhost).
+    required: false
+    default: localhost
+    type: str
+  port:
+    type: int
+    description:
+      - The port on which the consul agent is running.
+    required: false
+    default: 8500
+  scheme:
+    description:
+      - The protocol scheme on which the consul agent is running.
+    required: false
+    default: http
+    type: str
+  token:
+    description:
+      - A management token is required to manipulate the roles.
+    type: str
+  validate_certs:
+    type: bool
+    description:
+      - Whether to verify the TLS certificate of the consul agent.
+    required: false
+    default: True
+requirements:
+  - requests
+'''
+
+EXAMPLES = """
+- name: Create a role with 2 policies
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role
+    policies:
+      - id: 783beef3-783f-f41f-7422-7087dc272765
+      - name: "policy-1"
+- name: Create a role with service identity
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-2
+    service_identities:
+      - name: web
+        datacenters:
+          - dc1
+- name: Create a role with node identity
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-3
+    node_identities:
+      - name: node-1
+        datacenter: dc2
+- name: Remove a role
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-3
+    state: absent
+"""
+
+RETURN = """
+role:
+    description: The role object
+    returned: success
+    type: dict
+    sample: |
```

Comment:
Your sample is a string, not a dictionary.
```suggestion
    sample:
```